src/main/java/com/conveyal/r5/rastercost/CustomCostField.java

```diff
@@ -59,33 +51,73 @@ public class CustomCostField implements CostField, Serializable {
+    private double sensitivityCoefficient;
+
+    // CustomCostFactors are used to calculate the custom cost addition "cost seconds" 
+    // consists of osmId as the key and custom cost factor as value
+    private HashMap<Long, Double> customCostFactors = null;
+
+    // Flag for allowing null's for custom cost edges.
+    // If this is false the routing will force custom cost values for each edge so
+    // the rouiting will crash if no value found for each edge traversed
+    // if set to true, the processed edge will get a custom cost value of 0
+    private boolean allowNullCustomCostEdges;
+
+    /** 
+    * base edge travel times by osmid
+    * these costs are without the custom cost addition
+    *
+    * Key is osmId, value is base traversal time in seconds
+    */
+    private HashMap<Long, Integer> baseTraveltimes = new HashMap<Long, Integer>();
+
+    /**
+     * custom cost travel times by osmid
+     * these costs are the custom cost addition costs which are added to the base traversal time
+     * they are treated as seconds but are actually costs or "cost seconds"
+     * these values are calculaeted using formula: basetraveltime * customcostfactor * sensitivitycoefficient
+     * and then added to the base traversal time for each edge
+     * 
+     * Key is osmId, value is addition custom cost time as costs (seconds)
+     */
+    private HashMap<Long, Integer> customCostAdditionalTraveltimes = new HashMap<Long, Integer>();
+
 
 /**
  * 
  * @param displayKey
- * displayKey currently not really used, implemented due to CostField interface
+ * usefull when using multiple custom cost instances for naming
+ * also implemented due to CostField interface
  * @param sensitivityCoefficient
  * sensitivityCoefficient can currently also be a negative number
  * the only restricting factor is that the routing logic in MultistageTraversalTimeCalculator
  * will fallback to a positive number 1 if the sum of basetraversaltime + all custom cost found for that edge
  * will be less than 1. This is because the routing logic will not allow negative traversal times.
- * @param customCostMap
- * customCostMap has osmId as the key and the custom cost seconds as the value
+ * @param customCostFactors
+ * customCostFactors has osmId as the key and the custom cost seconds as the value
+ * @param allowNullCustomCostEdges
+ * flag for allowing or disallowing null costs for osmids in the customCostFactors
+ * if this is false, every edge/osmid key needs to have a value in the HashMap, otheriwse will throw error
+ * if true, will just fallback to 0 additionalSeconds for each edge where customCost value not found
  */
-    public CustomCostField (String displayKey, double sensitivityCoefficient, HashMap<Long, Double> customCostMap) {
-        validateCustomCostMap(customCostMap);
+    public CustomCostField (String displayKey, double sensitivityCoefficient, HashMap<Long, Double> customCostFactors, boolean allowNullCustomCostEdges) {
```

Comment:
Can you still create an overloaded function that does not accept the customCostFactors parameter, but sets it to a default value (should that be false?)

Reply:
do you mean the allowNullCustomCostEdges param?

Reply:
yes I can, I actually made it but then removed it because I thought that it would force user to make the decision of accepting "empty osm values".

Reply:
added the overload constructor